e2e-nodejs/group-hotwallet-with-blockhash/test-hotwallet-with-blockhash.mjs

```diff
@@ -14,58 +27,88 @@ export async function main() {
-  const origin = 'https://localhost/login';
   const statement =
     'This is a test statement. You can put anything you want here.';
 
-  const TEST_BLOCKHASH =
-    '0xfe88c94d860f01a17f961bf4bdfb6e0c6cd10d3fda5cc861e805ca1240c58553';
+  const accessControlConditions = [
+    {
+      contractAddress: '',
+      standardContractType: '',
+      chain,
+      method: 'eth_getBalance',
+      parameters: [':userAddress', 'latest'],
+      returnValueTest: {
+        comparator: '>=',
+        value: '0',
+      },
+    },
+  ];
 
   // ==================== Test Logic ====================
 
   const litNodeClient = new LitJsSdk.LitNodeClient({
     litNetwork: 'cayenne',
   });
   await litNodeClient.connect();
+
   let nonce = litNodeClient.getLatestBlockhash();
+  console.log('Eth blockhash nonce- ', nonce);
 
   if (!nonce) {
-    console.log(
-      "Latest blockhash is undefined as the corr node changes hasn't been deployed"
-    );
-    nonce = TEST_BLOCKHASH;
+    fail('Latest Eth blockhash is undefined');
   }
 
-  const siweMessage = new siwe.SiweMessage({
-    domain,
-    address: wallet.address,
-    statement,
-    uri: origin,
-    version: '1',
-    chainId: '1',
-    nonce,
+  const authNeededCallback = async ({ resources, expiration, uri }) => {
+    const message = new SiweMessage({
+      domain,
+      address: wallet.address,
+      statement,
+      uri,
+      version: '1',
+      chainId: '1',
+      expirationTime: expiration,
+      resources,
+      nonce,
+    });
+
+    const toSign = message.prepareMessage();
+    const signature = await wallet.signMessage(toSign);
+
+    const authSig = {
+      sig: signature,
+      derivedVia: 'web3.eth.personal.sign',
+      signedMessage: toSign,
+      address: wallet.address,
+    };
+
+    return authSig;
+  };
+
+  const hashedEncryptedSymmetricKeyStr = await hashBytes({
+    bytes: new Uint8Array(accessControlConditions),
   });
 
-  const messageToSign = siweMessage.prepareMessage();
-  const signature = await wallet.signMessage(messageToSign);
+  const litResource = new LitAccessControlConditionResource(
+    hashedEncryptedSymmetricKeyStr
+  );
 
   // ==================== Post-Validation ====================
 
-  const recoveredAddress = ethers.utils.verifyMessage(messageToSign, signature);
-
-  const authSig = {
-    sig: signature,
-    derivedVia: 'web3.eth.personal.sign',
-    signedMessage: messageToSign,
-    address: recoveredAddress,
-  };
-
-  console.log(authSig);
+  // NOTE: `getSessionSigs` will fail if the nonce is not a valid Eth blockhash
+  const sessionSigs = await litNodeClient.getSessionSigs({
+    chain,
+    resourceAbilityRequests: [
+      {
+        resource: litResource,
+        ability: LitAbility.AccessControlConditionDecryption,
+      },
+    ],
+    authNeededCallback,
+  });
 
-  if (!authSig.signedMessage.includes(TEST_BLOCKHASH)) {
-    return fail("authSig doesn't contain the blockhash");
-  }
+  console.log(sessionSigs);
```

Comment:
do we do anything with the `sessionSigs` beside log it? we can remove this if we are not doing assertions.

Reply:
No but the idea is to check whether the sessionSigs is created correctly

Reply:
Right, can we add some assertions thats the `latestBlockhash` is included in the objects contained in the `sessionSigsMap`